.gitignore

```diff
@@ -1,4 +1,5 @@
 #ignore build results
+src/WCSimRootDict.cxx
```

Comment:
Shouldn't this `src/` directory be removed too?

Reply:
Not here. This is relly for the "src" directory (in case someone decides to switch between ".cc" and ".cxx").

Reply:
why does it matter if it's a .cc file or .cxx? It's just a difference in file extension convention but the contents of the files are the same?

Reply:
but your changes in ` CMakeLists.txt` move this file to base directory, so then shouldn't there be another line without `src/`?

Also, are you saying that people need to manually change `.cc` to `.cxx` in [GNUmakefile_root](https://github.com/WCTE/WCSim/blob/nuPRISM/develop/GNUmakefile_root)?

Reply:
@nickwp then should the extensions in `GNUmakefile*` and `CMakeLists.txt` be made the same? 

In what case do we use `make` vs `cmake`?

Reply:
The ".gitignore" is only for the "src" directory. The content of the "build" directory is completely separated (i.e. never in git).

Reply:
Then could you fix all `GNUmakefile*` too, so it's applied to all build methods.

Reply:
It doesn't really matter if the dictionary is ".cc" or ".cxx". I just think it would be a good idea to use the same name regardless of the building method.

Reply:
> In what case do we use make vs cmake?
It's just preference really. Both work fine for me.

Reply:
Sorry, I was a bit wrong in my statements.
It seems that ROOT (5 and 6) will always create dictionaries with a ".cxx" suffix when CMake is used (it is hardcoded in the "`ROOT_GENERATE_DICTIONARY`" function).
So, if one wants to make it consistent, one would need to modify the "`GNUmakefile*`" method (I don't use it), which needs to be switched from ".cc" to ".cxx" (in principle, this can be done as, inside of these files, the "`rootcint`" is explicitly called with the "`-f`" flag and the full dictionary's file name is given there).

Reply:
ok, could you please be the "one" to implement and test? (In principle, each PR should improve all relevant parts of code for which it's designed for, to avoid code divergences.)